src/kbmod/search/kernels/image_kernels.cu

```diff
@@ -66,33 +65,44 @@ __global__ void convolve_psf(int width, int height, float *source_img, float *re
     }
 }
 
-extern "C" void deviceConvolve(float *source_img, float *result_img, int width, int height, PSF &psf) {
+extern "C" void deviceConvolve(float *source_img, float *result_img, int width, int height,
+                               float *psf_kernel, int psf_radius) {
     if (width <= 0) throw std::runtime_error("Invalid width = " + std::to_string(width));
     if (height <= 0) throw std::runtime_error("Invalid height = " + std::to_string(height));
-    int psf_radius = psf.get_radius();
     if (psf_radius < 0) throw std::runtime_error("Invalid PSF radius = " + std::to_string(psf_radius));
 
     uint64_t n_pixels = width * height;
     int psf_dim = 2 * psf_radius + 1;
+    int psf_size = psf_dim * psf_dim;
+    
+    // Compute the PSF sum.
+    float psf_sum = 0.0;
+    for (int r = 0; r < psf_dim; ++r) {
+        for (int c = 0; c < psf_dim; ++c) {
+            psf_sum += psf_kernel[r * psf_dim + c];
+        }
+    }
```

Comment:
I think `eigen` has a builtin `.sum()` method for their arrays, but I also see the kernel is being passed in as a `float *`, so is that an issue with passing between C++ and cuda?

Reply:
Yes. CUDA code and eigen don't play well together. So I need to pass as a raw array of floats and manually do the sum.